General coding practices/Test Automation (Mobile) - element IDs.md

```diff
@@ -0,0 +1,98 @@
+**Chapters:**
+
+- [General](#general)
+- [iOS](#ios)
+- [Android](#android)
+	- [XML architecture](#xml-architecture)
+	- [Compose architecture](#compose-architecture)
+
+## General
+
+**For elements that don't have an ID, the developer should by default add IDs when working on a feature/screen** (e.g., buttons, input fields, list containers and items, titles, etc.). Additionally, if needed, the tester will make a list of elements with missing IDs so that they can be added for test automation purposes. If the tester makes a list with proposed IDs, the developer can accept the proposed one if it makes sense or discuss with the tester to name it differently.
+
+While implementing IDs, the developer should sync with the responsible tester who will be doing the automated tests for IDs and implementation limitations. There shouldn’t be any compromises to the quality of features and usability when implementing these IDs.
+
+### Naming
+
+The IDs added to elements should be **unique** (most importantly - be unique on a specific screen). For that purpose, **prefixes and/or suffixes** can be used to define the ID of the component more precisely. That will be especially helpful in the case when the component consists of other sub-components which will also have IDs added to them.
+
+*Example:* 
+
+<span style="display:block; border: 1px solid #e0e0e0; margin-top:15px; margin-bottom:15px; margin-left:auto; margin-right:auto; width:80%;">![Android Naming Example](/img/test_automation/TA_Naming.png)</span>
+
+Developers can follow naming conventions for their respective platform (*camelCase*, *snake_case*, or other).
+
+## iOS
+
+### Accessibility Identifier
+
+For UI testing purposes Apple provides us with a so-called [accessibilityIdentifier](https://developer.apple.com/documentation/uikit/uiaccessibilityidentification/1623132-accessibilityidentifier). It is meant to "be used to uniquely identify a UI element in the scripts we write using the UI Automation interfaces". Accessibility identifiers aren't accessed by e.g. VoiceOver. They are part of the `UIAccessibilityIdentification` protocol, which consists of methods that associate a **unique identifier** with elements in a user interface. The only strict requirement for adhering to this protocol is defining the `accessibilityIdentifier` property.
+
+`UIAccessibilityIdentification: var accessibilityIdentifier: String? { get set }`
```

Comment:
This as well, you can omit it

Reply:
Sorry, I wasn't specific - the paragraph was fine and necessary 😄 but the code example can be omitted. I've expected from GH to highlight only than line of code thingy but for some reason it highlighted the whole chapter. Sorry!

Reply:
No worries, I thought the same because I expected GH to state which line is affected by the comment. 😄 
I reverted the removal and updated the text as per other comments. It should be okay now.